src/uct/ib/ud/verbs/ud_verbs.c

```diff
@@ -207,6 +207,12 @@ uct_ud_verbs_ep_am_zcopy(uct_ep_h tl_ep, uint8_t id, const void *header,
     UCT_CHECK_IOV_SIZE(iovcnt, uct_ib_iface_get_max_iov(&iface->super.super) - 1,
                        "uct_ud_verbs_ep_am_zcopy");
 
+    if (iface->super.config.max_inline) {
```

Comment:
something is wier here.. is the max header length 0 in case of max_inline==0? if yes why and is it really have to be this way?

Reply:
it is odd indeed.

Reply:
it's not clear, what is exactly the reason for test failure?

Reply:
max_header is reported as 0, but UD is able to transfer header using non-inline way.
fixed by more correct initialization of am_hdr_size value, updated assert